### ISSUE TITLE
combineExpression Error when specifying chain

### ISSUE THREAD

**Issue by jreimertz** [ncborcherding/scRepertoire#318]
Hi Nick,

I've run into the issue where if I try to specify a chain to use in the `combineExpression()` function I get the following error:

![image](https://github.com/ncborcherding/scRepertoire/assets/119626411/f5e0b367-d6dd-4746-a311-5e5a5ee485bd)

This is how I've bee calling the function:

```
sce <- combineExpression(combined_BCR, 
                         sce, 
                         cloneCall="gene", 
                         chain = "IGH", 
                         proportion = TRUE)
```
Thank you in advance!

**Comment by ncborcherding** [comment]
Hey Justin, 

Happy to help - whats the output of ```sessionInfo()```? 

Can you tell me more about your experiment? How many samples in combine_BCR,etc?  That might help track down the problem.

Thanks,
Nick

**Comment by jreimertz** [comment]
Hi Nick,

Thanks for the prompt response. This is the output of `sessionInfo()`:

```
R version 4.3.2 (2023-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22631)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8    LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                           LC_TIME=English_United States.utf8    

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] lubridate_1.9.3             forcats_1.0.0               stringr_1.5.1               dplyr_1.1.4                 purrr_1.0.2                
 [6] readr_2.1.5                 tidyr_1.3.1                 tibble_3.2.1                tidyverse_2.0.0             ggraph_2.1.0               
[11] gtools_3.9.5                qs_0.25.7                   scales_1.3.0                circlize_0.4.15             Seurat_5.0.1               
[16] SeuratObject_5.0.1          sp_2.1-3                    scater_1.30.1               scuttle_1.12.0              SingleCellExperiment_1.24.0
[21] SummarizedExperiment_1.32.0 Biobase_2.62.0              GenomicRanges_1.54.1        GenomeInfoDb_1.38.6         IRanges_2.36.0             
[26] S4Vectors_0.40.2            BiocGenerics_0.48.1         MatrixGenerics_1.14.0       matrixStats_1.2.0           BiocStyle_2.30.0           
[31] scRepertoire_2.0.0          ggplot2_3.4.4              

loaded via a namespace (and not attached):
  [1] spatstat.sparse_3.0-3     bitops_1.0-7              httr_1.4.7                RColorBrewer_1.1-3        tools_4.3.2              
  [6] sctransform_0.4.1         utf8_1.2.4                R6_2.5.1                  lazyeval_0.2.2            uwot_0.1.16              
 [11] withr_3.0.0               gridExtra_2.3             progressr_0.14.0          quantreg_5.97             cli_3.6.2                
 [16] spatstat.explore_3.2-6    fastDummies_1.7.3         iNEXT_3.0.0               labeling_0.4.3            spatstat.data_3.0-4      
 [21] ggridges_0.5.6            pbapply_1.7-2             stringdist_0.9.12         parallelly_1.36.0         VGAM_1.1-9               
 [26] rstudioapi_0.15.0         generics_0.1.3            shape_1.4.6               RApiSerialize_0.1.2       ica_1.0-3                
 [31] spatstat.random_3.2-2     vroom_1.6.5               Matrix_1.6-5              ggbeeswarm_0.7.2          fansi_1.0.6              
 [36] abind_1.4-5               lifecycle_1.0.4           yaml_2.3.8                SparseArray_1.2.4         Rtsne_0.17               
 [41] grid_4.3.2                promises_1.2.1            crayon_1.5.2              miniUI_0.1.1.1            lattice_0.22-5           
 [46] beachmat_2.18.0           cowplot_1.1.3             pillar_1.9.0              knitr_1.45                rjson_0.2.21             
 [51] future.apply_1.11.1       codetools_0.2-19          leiden_0.4.3.1            glue_1.7.0                data.table_1.15.0        
 [56] vctrs_0.6.5               png_0.1-8                 spam_2.10-0               gtable_0.3.4              xfun_0.41                
 [61] S4Arrays_1.2.0            mime_0.12                 tidygraph_1.3.1           survival_3.5-7            ellipsis_0.3.2           
 [66] fitdistrplus_1.1-11       ROCR_1.0-11               nlme_3.1-164              bit64_4.0.5               RcppAnnoy_0.0.22         
 [71] evd_2.3-6.1               irlba_2.3.5.1             vipor_0.4.7               KernSmooth_2.23-22        colorspace_2.1-0         
 [76] tidyselect_1.2.0          bit_4.0.5                 compiler_4.3.2            BiocNeighbors_1.20.2      SparseM_1.81             
 [81] ggdendro_0.1.23           DelayedArray_0.28.0       plotly_4.10.4             stringfish_0.16.0         lmtest_0.9-40            
 [86] digest_0.6.34             goftest_1.2-3             spatstat.utils_3.0-4      rmarkdown_2.25            XVector_0.42.0           
 [91] htmltools_0.5.7           pkgconfig_2.0.3           sparseMatrixStats_1.14.0  fastmap_1.1.1             rlang_1.1.3              
 [96] GlobalOptions_0.1.2       htmlwidgets_1.6.4         shiny_1.8.0               DelayedMatrixStats_1.24.0 farver_2.1.1             
[101] zoo_1.8-12                jsonlite_1.8.8            BiocParallel_1.36.0       BiocSingular_1.18.0       RCurl_1.98-1.14          
[106] magrittr_2.0.3            GenomeInfoDbData_1.2.11   dotCall64_1.1-1           patchwork_1.2.0           munsell_0.5.0            
[111] Rcpp_1.0.12               evmix_2.12                viridis_0.6.5             reticulate_1.35.0         truncdist_1.0-2          
[116] stringi_1.8.3             ggalluvial_0.12.5         zlibbioc_1.48.0           MASS_7.3-60               plyr_1.8.9               
[121] parallel_4.3.2            listenv_0.9.1             ggrepel_0.9.5             deldir_2.0-2              graphlayouts_1.1.0       
[126] splines_4.3.2             hash_2.2.6.3              tensor_1.5                hms_1.1.3                 igraph_2.0.1.1           
[131] spatstat.geom_3.2-8       cubature_2.1.0            RcppHNSW_0.6.0            pkgload_1.3.4             reshape2_1.4.4           
[136] ScaledMatrix_1.10.0       evaluate_0.23             RcppParallel_5.1.7        BiocManager_1.30.22       tzdb_0.4.0               
[141] tweenr_2.0.2              httpuv_1.6.14             MatrixModels_0.5-3        RANN_2.6.1                polyclip_1.10-6          
[146] future_1.33.1             scattermore_1.2           ggforce_0.4.1             rsvd_1.0.5                xtable_1.8-4             
[151] RSpectra_0.16-1           later_1.3.2               viridisLite_0.4.2         gsl_2.1-8                 beeswarm_0.4.0           
[156] cluster_2.1.6             timechange_0.3.0          globals_0.16.2          
```

There are 12 samples in combine_BCR all of which include barcodes also found in the SingleCellExperiment object, sce. It's also worth noting that combineExpression works fine for my dataset so long as I don't specify a chain. Please let me know if there's any other information that would be helpful

**Comment by ncborcherding** [comment]
Hey Justin,

That helped immensely - pushed a fix in the dev version that fixes the error. I have a list of things to test today and I'll make a pull request to the master branch and update this thread when it is ready to go.

One thing I noticed was that if you are using a chain argument in ```combineExpression()``` you will lose the IGL/K chain in your single-cell object. 

Although that is by design - it is not ideal. I have made a second issue to track that.
